src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,205 @@
+        System.out.println("Sorry I don't understand that! Can you rephrase?");
+        System.out.println(HORIZONTAL_LINE);
+    }
+
+    private static void printList() {
+        System.out.println(HORIZONTAL_LINE);
+        System.out.println("Here are the tasks in your list:");
+        for (int i = 0; i < numberOfTasks; i++) {
+            System.out.println((i + 1) + "." + tasks[i]);
+        }
+        System.out.println(HORIZONTAL_LINE);
+    }
+
+    private static void printDone(int taskNumber) {
+        System.out.println(HORIZONTAL_LINE);
+        System.out.println("Okie! Marked this as done:");
+        System.out.println((taskNumber + 1) + "." + tasks[taskNumber]);
+        System.out.println(HORIZONTAL_LINE);
+    }
+
+    private static void makeDone(String line) {
+        try {
+            handleDone(line);
+        } catch (NumberFormatException e) {
+            printMissingDoneIndexMsg();
+        } catch (NullPointerException e) {
+            printInvalidTaskIndexMsg();
+        }
+    }
+
+    private static void handleDone(String line) {
+        int taskNumber = Integer.parseInt(line.substring(END_OF_DONE_INDEX).trim()) - 1;
+        if (tasks[taskNumber].isDone()) {
+            System.out.println("This task is already done!");
+        } else {
+            tasks[taskNumber].markAsDone();
+            printDone(taskNumber);
+        }
+    }
+
+    private static void printInvalidTaskIndexMsg() {
+        System.out.println(HORIZONTAL_LINE);
+        System.out.println("Oh no! There isn't a task with that index");
+        System.out.println("Please try again!");
+        System.out.println(HORIZONTAL_LINE);
+    }
+
+    private static void printMissingDoneIndexMsg() {
+        System.out.println(HORIZONTAL_LINE);
+        System.out.println("Oh no! An integer must come after the done command!");
+        System.out.println("Please try again!");
+        System.out.println(HORIZONTAL_LINE);
+    }
+
+    private static void printTaskConfirmation() {
+        System.out.println(HORIZONTAL_LINE);
+        System.out.println("Umm ok added:");
+        System.out.println("  " + tasks[numberOfTasks - 1]);
+        System.out.println("Now you have " + numberOfTasks + " tasks in the list.");
+        System.out.println(HORIZONTAL_LINE);
+    }
+
+    private static void addEventTask(String line) {
+        try {
+            handleEventTask(line);
+            printTaskConfirmation();
+        } catch (StringIndexOutOfBoundsException e) {
+            printMissingAtErrorMsg();
+        }
+    }
+
+    private static void handleEventTask(String line) {
+        int endOfDescriptionIndex = line.indexOf("/at");
+        int startOfAtIndex = line.indexOf("/at") + AT_LENGTH;
+        String description = line.substring(END_OF_EVENT_INDEX, endOfDescriptionIndex).trim();
+        String at = line.substring(startOfAtIndex).trim();
+
+        tasks[numberOfTasks] = new Event(description, at);
+        numberOfTasks++;
+    }
```

Comment:
Do remember to add javadoc comments down the line for more complicated methods that require some explanation